### PR TITLE
doc: extensions: inventory_builder: support incremental builds

### PR DESCRIPTION
**Pull request by gmarull**
With these changes `inventory_builder` supports incremental builds, i.e. only processes modified files. Also added type hints to the extension and moved to a single file module.

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
LGTM

File: doc/_extensions/inventory_builder.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    name = 'inventory'
+    epilog = 'Only generates an inventory file.'
+
+    allow_parallel = True
+
+    def init(self) -> None:
+        pass
+
+    def get_outdated_docs(self) -> Iterator[str]:
+        for doc_name in self.env.found_docs:
```
Comment (mbolivar-nordic) [inline doc/_extensions/inventory_builder.py:33]: I didn't review the earlier PRs in detail so I'm just asking for my own knowledge: how does this work?

I don't see 'env' in the Builder API ref here: https://www.sphinx-doc.org/en/master/extdev/builderapi.html Not sure if that is the right place to look, though.

**Review by mbolivar-nordic — APPROVED** [review]
Seems fine, though I don't really know the details here

File: doc/_extensions/inventory_builder.py (hunk 1 above)
Comment (gmarull) [reply doc/_extensions/inventory_builder.py:33]: `self.env` contains an instance of `BuildEnvironment`, ref. https://www.sphinx-doc.org/en/master/extdev/envapi.html#sphinx.environment.BuildEnvironment. Other builder support this feature (HTML, XML...) see, e.g. https://github.com/sphinx-doc/sphinx/blob/4.x/sphinx/builders/xml.py#L46

**Pull request merged**